File: algorithms/1_easy/M/03402_minimum_operations_to_make_columns_strictly_increasing/solution.c
int minimumOperations(int *restrict *restrict grid, const int rows, const int *restrict cols_data) {
    if (rows <= 1 || grid == NULL || cols_data == NULL) { return 0; }

    const int cols = cols_data[0];
    int sum = 0;

    for (int j = 0; j < cols; j += 1) {
        int prev = grid[0][j];
        for (int i = 1; i < rows; i+= 1) {
            const int target = prev + 1;
            sum += MAX(0, target - grid[i][j]);
            prev = MAX(target, grid[i][j]);
        }
    }

    return sum;
}

File: algorithms/1_easy/M/03402_minimum_operations_to_make_columns_strictly_increasing/solution.rs
impl Solution {
    pub fn minimum_operations(grid: Vec<Vec<i32>>) -> i32 {
        if grid.len() <= 1 { return 0; }

        let (rows, cols) = (grid.len(), grid[0].len());
        let mut sum = 0;

        for j in 0..cols {
            let mut prev = grid[0][j];
            for i in 1..rows {
                let target = prev + 1;
                sum += std::cmp::max(0, target - grid[i][j]);
                prev = target.max(grid[i][j]);
            }
        }

        sum
    }
}
